src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/controller/sw/SwitchFsm.java

```diff
@@ -539,7 +539,13 @@ private void persistSwitchConnections(Switch sw, SwitchAvailabilityData availabi
 
         for (SwitchAvailabilityEntry entry : goal) {
             ConnectReference ref = new ConnectReference(entry);
-            persistSingleSwitchConnect(sw, entry, extra.remove(ref));
+            SwitchConnect connect = extra.get(ref);
+            try {
+                persistSingleSwitchConnect(sw, entry, connect);
+                extra.remove(ref);
+            } catch (IllegalArgumentException e) {
```

Comment:
Maybe we should use more precise exeption (or create new one)? All illegal argument exeptions will be logged in this way and we will lose stack trace for this kind of exceptions.

Reply:
We can pass a `Throwable` object to the `log.error` method.

Reply:
I have created custom exception... in some places it forced to make a mess.